autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
         sampler, _ = get_split_samplers(dataset, split_ratio=0.2)
         return SampledDataset(dataset, sampler)
 
-class TSVClassificationTask(AbstractGlueTask):
-    def __init__(self, *args, **kwargs): # passthrough arguments to TSVDataset
-        # (filename, field_separator=nlp.data.Splitter(','), num_discard_samples=1, field_indices=[2,1])
-        self.args = args
-        self.kwargs = kwargs
-        is_pair = False
-        class_labels = ['0', '1']
-        metric = CompositeEvalMetric()
-        metric.add(F1())
-        metric.add(Accuracy())
-        super(TSVClassificationTask, self).__init__(class_labels, metric, is_pair)
-        dataset = nlp.data.TSVDataset(*self.args, **self.kwargs)
-        # do the split
-        train_sampler, val_sampler = get_split_samplers(dataset, split_ratio=0.8)
-        self.trainset = SampledDataset(dataset, train_sampler)
-        self.valset = SampledDataset(dataset, val_sampler)
+class CustomTSVClassificationTask(AbstractGlueTask):
+    """
+
+    Parameters
+    ----------
+    filepath: str, path of the file
```

Comment:
how about multiple files?

Reply:
If Tabular's load_pd.load() was used it would be able to handle multiple files without issue.

Reply:
I am targeting on basic and popular case in this PR which can be found in most benchmark and competition, where each dataset is a single file. In future, we could support this. To leverage load_pd, we would need to refactory the code to make it compatible between modules.

Reply:
Why do you need to refactor load_pd? It is a file loading function. It does not need to be refactored.

Reply:
you could easily just extend load_pd to accept kwargs and pass those to pd.read_csv (if needed)

Reply:
three simple reasons: 1) load_pd is a tabular functionality in design and structure; 2) it can be refined from both implementation and documentation perspective; 3) minor thing: `load_pd` doesn't follow python naming practice.